crates/core/c8y_api/src/http_proxy.rs

```diff
@@ -89,29 +89,19 @@ impl C8yEndPoint {
         url_event_binary
     }
 
-    pub fn url_is_in_my_tenant_domain(&self, url: &str) -> bool {
+    pub fn possible_tenant_url(&self, url: &str) -> Option<Url> {
```

Comment:
The naming convention we have on thin-edge is to use the prefix `maybe` for optional values.

```suggestion
    pub fn maybe_tenant_url(&self, url: &str) -> Option<Url> {
```